### ISSUE TITLE
Add turncost for uphill and downhill

### ISSUE THREAD

**Issue by carg0cult** [abrensch/brouter#493]
When cycling, it makes a huge difference whether a turn is uphill or downhill. It would be great if we had separate turncost for uphill, downhill and no hill.

**Comment by polyscias** [comment]
Turncost are only calculated for crossings so not for hairpin bends and other bends.

I think that keeping this in mind the difference in uphill versus downhill cost is small, agreed?

**Comment by quaelnix** [comment]
@polyscias, that's not true. Turncosts are determined each time the direction of travel changes. This [continuous road segment](https://www.openstreetmap.org/way/31249681) does not cross any other way and changing the `turncost` from `0` to `100` in the ["Shortest" profile](https://brouter.de/brouter-web/#map=17/51.79479/10.62045/standard&lonlats=10.621969,51.796535;10.61789,51.794138&profile=shortest) leads to the following results:
| | turncost=0 | turncost=100 |
| :---: | ---: | ---: |
| Cost | 500 | 566  |

**Comment by poutnikl** [comment]
> @polyscias, that's not true. Turncosts are determined each time the direction of travel changes.

I have been once experimenting with an on demand profile for some motorbiker with Costfactor offset and negative turncost. He wanted to prefer winding ways to straight ones. It kind of worked, but it had to be conservative otherwise would occacionally cause routing error if the segment cost got negative.

**Comment by EssBee59** [comment]
> but it had to be conservative otherwise would occacionally cause routing error if the segment cost got negative.

yes, I had also a very big "surprise" with a profile where the cost got < 1 !!! (0.9)
